### PR TITLE
Worked on Login and Signup

### PR DESCRIPTION
**Pull request by rammyblog**

### CODE REVIEW

File: src/state/auth/authActionCreator.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+export const authLogin = (userData) => {
+  return function (dispatch) {
+    dispatch({ type: types.AUTH_START })
+
+    return axios
+      .post(`${apiURL}/login`, userData)
+      .then((res) => {
+        const token = res.data.key
+        const expirationDate = new Date(new Date().getTime() + 3600 * 1000)
+        localStorage.setItem("codeclan_token", token)
```
Comment (tayormi) [inline src/state/auth/authActionCreator.js:39]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

**Comment by tayormi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
- Added 134
           

Complexity increasing per file
==============================
- src/components/landing.js  1
- src/state/auth/authReducer.js  4
- src/state/auth/authActionCreator.js  1
- src/components/containers/PrivateRoute.js  2
- src/components/auth/Login.js  4
- src/components/auth/Signup.js  4
- src/components/dashboard.js  1
- src/routes.js  1
- src/components/auth/utils.js  1
         

Clones added
============
- src/components/auth/Login.js  2
- src/components/auth/Signup.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963&bid=18457109)

File: src/state/auth/history.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+//history.js
+import createHistory from "history/createBrowserHistory"
```
Comment (tayormi) [inline src/state/auth/history.js:2]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/index.js [hunk 3]
```diff
@@ -1,33 +1,28 @@
...
+import { createStore, compose, applyMiddleware } from "redux"
+import { Provider } from "react-redux" // providing your React App with redux
+import logger from "redux-logger"
+import thunk from "redux-thunk"
+import MonsterReducer from "./state/monsterReducer"
+import "./index.css"
+import App from "./App"
 
-const middleWare = [thunk, logger];
+const middleWare = [thunk, logger]
```
Comment (tayormi) [inline src/index.js:11]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/index.js [hunk 4]
```diff
@@ -1,33 +1,28 @@
...
-import logger from "redux-logger";
-import thunk from "redux-thunk";
-import MonsterReducer from "./state/monsterReducer";
-import "./index.css";
-import App from "./App";
+import React from "react"
+import ReactDOM from "react-dom"
+import { createStore, compose, applyMiddleware } from "redux"
+import { Provider } from "react-redux" // providing your React App with redux
+import logger from "redux-logger"
```
Comment (tayormi) [inline src/index.js:5]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/state/auth/history.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+
+const history = createHistory()
+
+export default history
```
Comment (tayormi) [inline src/state/auth/history.js:6]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/components/auth/Signup.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+  const [values, setValues] = useState({
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+  })
+
+  const [emailError, setEmailError] = useState(false)
+  const handleSubmit = (e) => {
+    e.preventDefault()
```
Comment (tayormi) [inline src/components/auth/Signup.js:17]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/components/auth/css/signup.css [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+  justify-content: center;
+}
+.signIn span {
+  padding-left: 0.3rem;
+}
+span a {
+  color: darkblue;
+  font-weight: 500;
+}
+input {
```
Comment (tayormi) [inline src/components/auth/css/signup.css:59]: Codacy Issue found: [Expected empty line before rule (rule-empty-line-before)](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/index.js [hunk 8]
```diff
@@ -1,33 +1,28 @@
...
 
 export const store = createStore(
   MonsterReducer,
   {},
-  composeEnhancers(
-    applyMiddleware(...middleWare),
-  )
-);
+  composeEnhancers(applyMiddleware(...middleWare))
+)
```
Comment (tayormi) [inline src/index.js:19]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/components/auth/Login.js [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+import LoginStyled from "./LoginStyled"
+function LoginForm({ authLogin, loading, error, errResponse, token, history }) {
+  const [values, setValues] = useState({
+    email: "",
+    password: "",
+  })
+
+  const [emailError, setEmailError] = useState(false)
+  const handleSubmit = (e) => {
+    e.preventDefault()
```
Comment (tayormi) [inline src/components/auth/Login.js:16]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/state/auth/authActionCreator.js [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+      .then((res) => {
+        localStorage.setItem("codeclan_token", res.data)
+        const expirationDate = new Date(new Date().getTime() + 3600 * 1000)
+        localStorage.setItem("expirationDate", expirationDate)
+
+        dispatch({ type: types.AUTH_SUCCESS, payload: res.data })
+        // history.push(`/dashboard`)
+      })
+      .catch((err) => {
+        console.log(err, err.response)
```
Comment (tayormi) [inline src/state/auth/authActionCreator.js:20]: Codacy Issue found: [Unexpected console statement.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/components/auth/Login.js [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+    authLogin(values)
+  }
```
Comment (tayormi) [inline src/components/auth/Login.js:18]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/components/containers/PrivateRoute.js [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+      localStorage.getItem("user") ? (
+        <Component {...props} />
+      ) : (
+        <Redirect
+          to={{ pathname: "/login", state: { from: props.location } }}
+        />
+      )
+    }
+  />
+)
```
Comment (tayormi) [inline src/components/containers/PrivateRoute.js:17]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/components/auth/Login.js [hunk 13]
```diff
@@ -0,0 +1,105 @@
+import React, { useState, useEffect } from "react"
+import { authLogin } from "../../state/auth/authActionCreator"
+import { connect } from "react-redux"
+// import "./css/signin.css"
+import { validateEmail } from "./utils"
+import { Link } from "react-router-dom"
```
Comment (tayormi) [inline src/components/auth/Login.js:6]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/components/auth/SignupStyled.js [hunk 14]
```diff
@@ -0,0 +1,108 @@
+import styled from "styled-components"
```
Comment (tayormi) [inline src/components/auth/SignupStyled.js:1]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/components/auth/Login.js [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+    error: store.auth.error,
+    errResponse: store.auth.errResponse,
+  }
+}
+
+const mapDispatchToProps = {
+  authLogin,
+}
+
+export default connect(mapStateToProps, mapDispatchToProps)(LoginForm)
```
Comment (tayormi) [inline src/components/auth/Login.js:105]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/components/auth/Signup.js [hunk 16]
```diff
@@ -0,0 +1,128 @@
...
+    register(values)
```
Comment (tayormi) [inline src/components/auth/Signup.js:18]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/components/auth/Login.js [hunk 17]
```diff
@@ -0,0 +1,105 @@
+import React, { useState, useEffect } from "react"
+import { authLogin } from "../../state/auth/authActionCreator"
+import { connect } from "react-redux"
+// import "./css/signin.css"
+import { validateEmail } from "./utils"
+import { Link } from "react-router-dom"
+import LoginStyled from "./LoginStyled"
```
Comment (tayormi) [inline src/components/auth/Login.js:7]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/routes.js [hunk 18]
```diff
@@ -0,0 +1,18 @@
+import React from "react"
```
Comment (tayormi) [inline src/routes.js:1]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/components/auth/Login.js [hunk 19]
```diff
@@ -0,0 +1,105 @@
...
+function LoginForm({ authLogin, loading, error, errResponse, token, history }) {
+  const [values, setValues] = useState({
+    email: "",
+    password: "",
+  })
+
+  const [emailError, setEmailError] = useState(false)
+  const handleSubmit = (e) => {
+    e.preventDefault()
+    authLogin(values)
```
Comment (tayormi) [inline src/components/auth/Login.js:17]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/index.css [hunk 20]
```diff
@@ -1,13 +1,27 @@
+@import-normalize; /* bring in normalize.css styles */
```
Comment (tayormi) [inline src/index.css:1]: Codacy Issue found: [Unknown @ rule: @import-normalize.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/components/auth/css/signup.css [hunk 21]
```diff
@@ -0,0 +1,120 @@
...
+  text-transform: capitalize;
+  font-weight: 400;
+}
+.info {
+  font-size: 13px;
+  font-weight: 600;
+  text-transform: capitalize;
+}
+.signUp {
+  padding: 0rem 0.2rem;
```
Comment (tayormi) [inline src/components/auth/css/signup.css:21]: Codacy Issue found: [Unexpected unit (length-zero-no-unit)](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/components/auth/css/signup.css [hunk 22]
```diff
@@ -0,0 +1,120 @@
...
+  color: #1f59bb;
+  text-transform: capitalize;
+  font-weight: 400;
+}
+.info {
+  font-size: 13px;
+  font-weight: 600;
+  text-transform: capitalize;
+}
+.signUp {
```
Comment (tayormi) [inline src/components/auth/css/signup.css:20]: Codacy Issue found: [Expected empty line before rule (rule-empty-line-before)](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/components/auth/Login.js [hunk 23]
```diff
@@ -0,0 +1,105 @@
+import React, { useState, useEffect } from "react"
+import { authLogin } from "../../state/auth/authActionCreator"
+import { connect } from "react-redux"
```
Comment (tayormi) [inline src/components/auth/Login.js:3]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/App.js [hunk 24]
```diff
@@ -1,14 +1,16 @@
-import React from "react";
-import Signup from "./components/Signup";
-import "./App.css";
+import React from "react"
+import { BrowserRouter as Router } from "react-router-dom"
```
Comment (tayormi) [inline src/App.js:2]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

File: src/index.js [hunk 25]
```diff
@@ -1,33 +1,28 @@
...
-import { Provider } from "react-redux"; // providing your React App with redux
-import { BrowserRouter as Router } from "react-router-dom";
-import logger from "redux-logger";
-import thunk from "redux-thunk";
-import MonsterReducer from "./state/monsterReducer";
-import "./index.css";
-import App from "./App";
+import React from "react"
+import ReactDOM from "react-dom"
+import { createStore, compose, applyMiddleware } from "redux"
```
Comment (tayormi) [inline src/index.js:3]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/gh/codeclannigeria/codeclannigeria-frontend/pullRequest?prid=5635963)

**Comment by rammyblog** [comment]
The main issue here is semicolon my vscode is not setup that way

**Comment by ofuochi** [comment]
@rammyblog, it's not  VSCode, it's your prettier that's auto-fixing it. You can overwrite the setting in the project by adding `.prettierrc` file and adding your settings

**Pull request merged**